rust/common/src/packed_func.rs

```diff
@@ -15,298 +17,305 @@ pub trait PackedFunc =
+                        TVMTypeCode_kTVMType => Type($value.v_type),
+                        TVMTypeCode_kTVMContext => Context($value.v_ctx),
+                        TVMTypeCode_kHandle => Handle($value.v_handle),
+                        TVMTypeCode_kArrayHandle => ArrayHandle($value.v_handle as TVMArrayHandle),
+                        TVMTypeCode_kNodeHandle => NodeHandle($value.v_handle),
+                        TVMTypeCode_kModuleHandle => ModuleHandle($value.v_handle),
+                        TVMTypeCode_kFuncHandle => FuncHandle($value.v_handle),
+                        TVMTypeCode_kNDArrayContainer => NDArrayContainer($value.v_handle),
+                        $( $tvm_type => { $from_tvm_type } ),+
+                        _ => unimplemented!("{}", type_code),
+                    }
+                }
+            }
+
+            pub fn into_tvm_value(self) -> (TVMValue, TVMTypeCode) {
+                use $name::*;
+                match self {
+                    Int(val) => (TVMValue { v_int64: val }, DLDataTypeCode_kDLInt),
+                    UInt(val) => (TVMValue { v_int64: val as i64 }, DLDataTypeCode_kDLUInt),
+                    Float(val) => (TVMValue { v_float64: val }, DLDataTypeCode_kDLFloat),
+                    Null => (TVMValue{ v_int64: 0 },TVMTypeCode_kNull),
+                    Type(val) => (TVMValue { v_type: val }, TVMTypeCode_kTVMType),
+                    Context(val) => (TVMValue { v_ctx: val }, TVMTypeCode_kTVMContext),
+                    String(val) => {
+                        (
+                            TVMValue { v_handle: val.into_raw() as *mut c_void },
+                            TVMTypeCode_kStr,
+                        )
+                    }
+                    Handle(val) => (TVMValue { v_handle: val }, TVMTypeCode_kHandle),
+                    ArrayHandle(val) => {
+                        (
+                            TVMValue { v_handle: val as *const _ as *mut c_void },
+                            TVMTypeCode_kArrayHandle,
+                        )
+                    },
+                    NodeHandle(val) => (TVMValue { v_handle: val }, TVMTypeCode_kNodeHandle),
+                    ModuleHandle(val) =>
+                        (TVMValue { v_handle: val }, TVMTypeCode_kModuleHandle),
+                    FuncHandle(val) => (TVMValue { v_handle: val }, TVMTypeCode_kFuncHandle),
+                    NDArrayContainer(val) =>
+                        (TVMValue { v_handle: val }, TVMTypeCode_kNDArrayContainer),
+                    $( $self_type($val) => { $from_self_type } ),+
+                }
+            }
+        }
+    }
 }
 
-impl<'a> TVMArgValue<'a> {
-    pub fn new(value: TVMValue, type_code: i64) -> Self {
-        TVMArgValue {
-            _lifetime: PhantomData,
-            value: value,
-            type_code: type_code,
+TVMPODValue! {
+    /// A borrowed TVMPODValue. Can be constructed using `into()` but the preferred way
```

Comment:
For docs to be generated, you need to capture them as `#[doc=$doc]` like [this](https://github.com/rust-ndarray/ndarray/blob/master/src/impl_ops.rs#L51) for example.

Reply:
Rustc makes a [docs pass](https://doc.rust-lang.org/rustdoc/passes.html#passes) before expanding macros that condenses all of the doc comments into `#[doc(blah)]`